lib/network_utils.ml

```diff
@@ -1163,3 +1200,196 @@ module Ethtool = struct
+		ignore(call_script ~timeout:(Some !rebuild_initrd_timeout) dracut 
+			["-f"; Printf.sprintf "/boot/initrd-%s.img" img_name; img_name])
+	
+	let gen_options_for_maxvfs driver max_vfs =
+		let gen_list value repeat = 
+			let rec aux acc = function
+				| n when n <= 0 -> acc
+				| n -> aux (value :: acc) (n-1)
+			in
+			aux [] repeat
+		in
+		Sysfs.get_dev_nums_with_same_driver driver
+		|> gen_list (string_of_int max_vfs)
+		|> String.concat ","
+
+	(* For given driver like igb, we parse each line of igb.conf which is the modprobe
+	configuration for igb. We keep the same the lines that do not have SRIOV configurations and 
+	change lines that need to be changed with patterns like `options igb max_vfs=4`
+	*)
+	let parse_modprobe_conf_internal file_path driver option =
+		(* Initially I did not use ref here, but afterward changed to ref to make the code easy to read *)
+		let has_probe_conf = ref false in
+		let need_rebuild_initrd = ref false in
+		let parse_single_line s = 
+			let parse_driver_options s = 
+				match Xstringext.String.split ~limit:2 '=' s with
+				(* has SRIOV configuration but the max_vfs is exactly what we want to set, so no changes and return s *)
+				| [k; v] when k = "max_vfs" && v = option ->  has_probe_conf := true; s
+				(* has SRIOV configuration and we need change it to expected option *)
+				| [k; v] when k = "max_vfs"  -> 
+					has_probe_conf := true;
+					need_rebuild_initrd := true;
+					debug "change SRIOV options from [%s=%s] to [%s=%s]" k v k option;
+					Printf.sprintf "max_vfs=%s" option
+				(* we do not care the lines without SRIOV configurations *)
+				| _ -> s
+			in
+			let trimed_s = String.trim s in
+			if Re.execp (Re_perl.compile_pat ("options[ \t]+" ^ driver)) trimed_s then
+				let driver_options = Re.split (Re_perl.compile_pat "[ \t]+") trimed_s in
+				List.map parse_driver_options driver_options
+				|> String.concat " "
+			else
+				trimed_s
+		in
+		let lines = try Unixext.read_lines file_path with _ -> [] in
+		let new_conf = List.map parse_single_line lines in
+		!has_probe_conf, !need_rebuild_initrd, new_conf
+
+	(*
+	returns ( a * b * c) where
+	a indicates the probe configuration already has the max_vfs options, meaning the device doesn't support sysfs and will be configed by modprobe
+	b indicates some changes shall be made on the coniguration to enable sriov to max_vfs, so we shall rebuild the initrd.
+	and c is the configurations after these changes
+	*)
+	let parse_modprobe_conf driver max_vfs =
+		let file_path = Printf.sprintf "/etc/modprobe.d/%s.conf" driver in
+		let option = gen_options_for_maxvfs driver max_vfs in
+		parse_modprobe_conf_internal file_path driver option
+
+	let get_sriov_info lspci_output =
+		if Re.execp (Re_perl.compile_pat "Capabilities:.+SR-IOV") lspci_output then
+		begin
+			let group = Re.exec_opt (Re_perl.compile_pat "Total VFs: (\d+), Number of VFs: (\d+)") lspci_output in
+			match group with
+			| None -> {max_vfs=0; num_vfs=0}
```

Comment:
Using a record for returning to simple integers is a bit overkill. A tuple is fine. However, in this case I had expected two functions: `Sysfs.sriov_numvfs` and `Sysfs.sriov_totalvfs`.